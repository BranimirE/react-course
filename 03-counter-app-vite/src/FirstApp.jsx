import PropTypes from 'prop-types';
export const FirstApp = ({title, subtitle}) => {
  return (
    <>
      <h1>{title}</h1>
      <h3>{subtitle}</h3>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
};

FirstApp.defaultProps = {
  // title: 'There is not title',
  subtitle: 'There is not subtitle'
}
