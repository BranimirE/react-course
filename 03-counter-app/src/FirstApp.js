import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ greeting, subtitle }) => {
    return (
        <>
            <h1>{greeting}</h1>
            <h3>{subtitle}</h3>
        </>
    )
}
FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'This is a sub-title'
}

export default FirstApp;