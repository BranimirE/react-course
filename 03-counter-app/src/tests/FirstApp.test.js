import { shallow } from 'enzyme';
import FirstApp from '../FirstApp';
//import { render } from "@testing-library/react";

describe('Tests in <FirstApp /> component', () => {
    test('Should show the greeting', () => {
        /* const greeting = 'Hello, I am Branimir';
        const { getByText } = render(<FirstApp greeting={greeting} />);
        expect(getByText(greeting)).toBeInTheDocument(); */
    });


    test('Should show <FirstApp /> correctly', () => {
        const greeting = 'Hello, I am Branimir';
        const wrapper = shallow(<FirstApp greeting={ greeting } />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show the text passed as prop', () => {
        const greeting = 'Hello, I am Branimir';
        const subTitle = 'This is a new subtitle';
        const wrapper = shallow(
            <FirstApp 
                greeting={ greeting } 
                subtitle={ subTitle }
            />);

        const h3Text = wrapper.find('h3').text();
        expect(h3Text).toBe(subTitle);
    });
});