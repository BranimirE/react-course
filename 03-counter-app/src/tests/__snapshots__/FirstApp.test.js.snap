// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> component Should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Branimir
  </h1>
  <h3>
    This is a sub-title
  </h3>
</Fragment>
`;
